app/models/invitation.rb

```diff
@@ -13,6 +13,8 @@ class InvitationAlreadyUsed < StandardError
   belongs_to :invitable, polymorphic: true
   belongs_to :canceller, class_name: User
 
+  update_counter_cache :invitable, :invitations_count
```

Comment:
Yea, I am also keeping an eye out for if these are good value.

It adds hooks for create, update and destroy. So there is a count query per invitation operation. It's a bigger cost that what you see with an increment/decrement approach but it's less problematic.